lib/wal/wal_buffer.c

```diff
@@ -532,7 +532,7 @@ wal_bufset_curoff(struct wal_bufset *wbs, int offc, uint64_t *offv)
 int
 wal_bufset_flushoff(struct wal_bufset *wbs, int offc, uint64_t *offv)
 {
-    assert(offc >= wbs->wbs_bufc);
+    INVARIANT(offc >= wbs->wbs_bufc);
```

Comment:
The weird thing about this invariant is that if offc is less than zero it won't trigger (cuz offc will be coerced to an unsigned and then compared with bufc which will be wrong, but won't trigger an assert).  Similarly, the (offc < wbs->wbs_bufc) check will fail if offc is less than zero.

Reply:
yeah, the signature of this function is just wrong. The second argument must be an uint as it just a vector count.